### PR TITLE
Automatic updates.

### PR DESCRIPTION
**Pull request by peppy**
Using squirrel for the time being. May change in the future depending on how things work.

- [x] Merge https://github.com/ppy/osu-framework/pull/473 first for title support.

### CODE REVIEW

File: osu.Game/Overlays/Notifications/ProgressNotification.cs [hunk 1]
```diff
@@ -17,49 +17,96 @@ namespace osu.Game.Overlays.Notifications
...
         }
 
-        public ProgressNotificationState State
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            //we may have received changes before we were displayed.
+            State = state;
+            Progress = progress;
```
Comment (Tom94) [inline osu.Game/Overlays/Notifications/ProgressNotification.cs]: Do we also need a Text = text assignment here?

Comment (peppy) [inline osu.Game/Overlays/Notifications/ProgressNotification.cs]: nope, it's assigned directly in the children array

Comment (Tom94) [inline osu.Game/Overlays/Notifications/ProgressNotification.cs]: That's a race condition though. If loading happens off the main thread IsLoaded won't be true within `load` and after Text has been set in there.

File: osu.Game/Overlays/Notifications/SimpleNotification.cs [hunk 2]
```diff
@@ -12,12 +12,33 @@ namespace osu.Game.Overlays.Notifications
     public class SimpleNotification : Notification
     {
         private string text;
+        public string Text
```
Comment (Tom94) [inline osu.Game/Overlays/Notifications/SimpleNotification.cs]: Same reason applies to these, also need to be set in LoadComplete.

File: osu.Game/Overlays/Notifications/ProgressNotification.cs [hunk 3]
```diff
@@ -68,10 +115,7 @@ public ProgressNotificationState State
...
         private Color4 colourCancelled;
 
-        public ProgressNotification(string text)
-        {
-            this.text = text;
-        }
+        private SpriteText textDrawable;
 
         [BackgroundDependencyLoader]
         private void load(OsuColour colours)
```
Comment (Tom94) [inline osu.Game/Overlays/Notifications/ProgressNotification.cs]: Can you move everything that does not need `colours` into LoadComplete?

File: osu.Game/Database/BeatmapDatabase.cs [hunk 4]
```diff
@@ -83,64 +83,66 @@ public void Reset()
...
-                            beatmapSet.Beatmaps.Add(beatmap.BeatmapInfo);
-                        }
+                        beatmapSet.Beatmaps.Add(beatmap.BeatmapInfo);
                     }
                 }
-
-                Import(new[] { beatmapSet });
             }
+
+            Import(new[] { beatmapSet });
```
Comment (Tom94) [inline osu.Game/Database/BeatmapDatabase.cs]: Redundant array, now that you refactored

**Review by Tom94 — APPROVED** [review]

**Pull request merged**